rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java

```diff
@@ -225,20 +225,65 @@ public static Menu createContextMenuFromCompactView(RMController controller, Obj
 
     public void disableProviderItems(ContextMenu menu, String url) {
         LoginModel loginModel = LoginModel.getInstance();
-        if (!loginModel.userHasProviderPermissionForNodeSource(url)) {
+        boolean permissionForNode = loginModel.userHasProviderPermissionForNode(url);
+        if (!permissionForNode || !loginModel.userHasPermissionToLockNodes()) {
             disableLockMenuItem(menu);
+        }
+        if (!permissionForNode || !loginModel.userHasPermissionToUnLockNodes()) {
             disableUnlockMenuItem(menu);
+        }
+        if (!permissionForNode || !hasMethodPermissionToRemoveItem(menu)) {
             disableRemoveMenuItem(menu);
         }
```

Comment:
Isn't testing `!loginModel.userHasPermission...()` and `!hasMethodPermissionToRemoveItem(menu)` methods sufficient ?

If you keep `!permissionForNode` test you can optimize it a little by not testing in each if.
For instance

```
if (!loginModel.userHasProviderPermissionForNode(url)){
	disableLockMenuItem(menu);
	disableUnlockMenuItem(menu);
	disableRemoveMenuItem(menu);
	return;
} else {
	if (!loginModel.userHasPermissionToLockNodes()) {
		disableLockMenuItem(menu);
	} 
	
	if (!loginModel.userHasPermissionToUnLockNodes()) {
		disableUnlockMenuItem(menu);
	}
	
	if (!hasMethodPermissionToRemoveItem(menu)) {
		disableRemoveMenuItem(menu);
	}
}
```